src/analysis/modals/DeleteEnvironment.ts

```diff
@@ -116,9 +128,6 @@ export const DeleteEnvironment = ({
         [
           !runtimeConfig && !!persistentDiskId,
           () => {
-            if (!deleteDiskSelected) {
-              setDeleteDiskSelected(true);
-            }
```

Comment:
After properly rendering `DeleteEnvironment` as a component, this produced an error:
> Warning: Cannot update a component (`AzureComputeModalBase`) while rendering a different component (`DeleteEnvironment`). To locate the bad setState() call inside `DeleteEnvironment`

Generally, `setState` shouldn't be inlined in render. Moving it to an effect resolved this error.